### ISSUE TITLE
Handle embark--restart in collect buffers

### ISSUE THREAD

**Issue by minad** [oantolin/embark#594]
I wonder if it would make sense to add some special handling for embark--restart to collect buffers. When I remove objects there it is slightly confusing that the candidates stay around. The idea would be to strike-through or remove the candidate which was acted upon. We could also implement this via a separate function embark--collect-crossout/remove/strikethrough. Also embark--restart should maybe renamed to embark--minibuffer-restart. However it may be better to use the export buffer for the remove use case anyway? What do you think?

**Comment by oantolin** [comment]
Well, I don't think this can reasonably be added to `embark–restart`, but we could add some post-action hooks for specific deletion or removal actions.

The thing is that the generic restart-anything behavior for embark collect would require rerunning the command the candidates came from, entering the input and then somehow figuring out when the candidates have finished coming in (in the case of async searches). In general there may not even be a finished candidate list, and any approach with timers would be a little brittle or take too long. Another option of course would be tighter coupling with consult for those async searches, to have consult somehow let embark register a callback for when the search is done, but we've always avoided those non-generic solution that require package coordination.

Now, what can be reasonably done is post action hooks for specific deletion or removal commands. That way the hook knows which item specifically need removing from the collect buffer.

**Comment by minad** [comment]
I meant something more primitive, just marking/removing the candidate directly to indicate the manipulation, but only in collect buffers.

**Comment by oantolin** [comment]
Yes, that's also what I thought you meant and I'm saying it can't be added to embark--restart but it can easily be made into a separate post action hook for certain actions.
